src/olympia/reviewers/views.py

```diff
@@ -1234,3 +1238,47 @@ def flags(self, request, **kwargs):
             ActivityLog.create(amo.LOG.ADMIN_ALTER_INFO_REQUEST, addon)
         serializer.save()
         return Response(serializer.data)
+
+
+class BrowseVersionViewSet(RetrieveModelMixin, GenericViewSet):
+    permission_classes = [AnyOf(
+        AllowReviewer, AllowReviewerUnlisted, AllowAddonAuthor,
+    )]
+
+    serializer_class = AddonBrowseVersionSerializer
+
+    def get_queryset(self):
+        """Return queryset to be used for the view."""
+        # Permission classes disallow access to non-public/unlisted add-ons
+        # unless logged in as a reviewer/addon owner/admin, so we don't have to
+        # filter the base queryset here.
+        return Version.unfiltered.all()
+
+    def get_serializer_context(self):
+        context = super(BrowseVersionViewSet, self).get_serializer_context()
+        context.update({
+            'file': self.request.GET.get('file', None)
+        })
+        return context
+
+    def check_object_permissions(self, request, obj):
+        """
+        Check if the request should be permitted for a given version.
+
+        This calls the DRF implementation but forwards the respective Add-on
+        as `obj` since most of our permission classes are based on an Add-on
+        being passed around.
+        """
+        # If the instance is marked as deleted and the client is not allowed to
+        # see deleted instances, we want to return a 404, behaving as if it
+        # does not exist.
+        can_see_deleted_addons = (
```

Comment:
By setting this in a variable early, you're making the query every time, even for non-deleted add-ons. Since it's just a single call I think it doesn't need to be a variable and could just be done in the condition below, where it'd be guarded by the `obj.deleted` part of the condition.